lib/IRGen/GenMeta.cpp

```diff
@@ -2745,10 +2745,11 @@ namespace {
           }
           assert(IGM.isKnownEmpty(Type.getFieldType(field, IGM.getSILModule()),
                                   ResilienceExpansion::Maximal));
-          B.addInt(IGM.SizeTy, 0);
+          B.addInt32(0);
         }
       };
       Scanner(IGM, Target, getLoweredType(), init).layout();
+      init.addAlignmentPadding(IGM.getPointerAlignment());
```

Comment:
@jckarter I'm facing a problem with this `addAlignmentPadding` since it adds `[4 x i8] zero initialized array` as an element which differs from of array element type (which results in failure in `finishArray`), is the a workaround for this? I guess there should be another way to get the end address of the array after it has been finalized, but i can't find one...

Reply:
`addAlignmentPadding` probably ought to be overridden by `ConstantArrayBuilder` to form alignment padding using the array element type instead of byte arrays.